src/compiler/parser.ts

```diff
@@ -2665,7 +2626,7 @@ namespace ts {
                 // for now we just check if the next token is an identifier.  More heuristics
                 // can be added here later as necessary.  We just need to make sure that we
                 // don't accidently consume something legal.
-                return lookAhead(nextTokenIsIdentifierOnSameLine);
+                return lookAhead(nextTokenIsIdentifierOrKeywordOrNumberOnSameLine);

```

Comment:
What about
```ts
for (yield of foo) { }
```
Do we do the right thing here? Same question for `yield instanceOf foo` and `yield in foo`. Does yield have low enough precedence to not even get here?